department-of-veterans-affairs/vets-api
# [65639] Configure retries for SignUpServiceUpdaterJob

## Summary

This PR configures the number of retries the SignUpServiceUpdaterJob will perform before failing to 15 (approximately 2.1 days). It also logs a warning message when retries have been exhausted to facilitate auditing.

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#65639

## Testing done

Unit tests cover the configuration of the worker itself.

Functional tests can be executed by:

* Induce the job to failure (ICN not represent in SuS, SuS unavailable, etc.)
* Keep the failure condition constant for 2.1 days.
* Observe that 15 retries have been attempted.
* Observe that the final error message has been logged.
* Observe that no further retries occur.

## Screenshots

Not relevant.

## What areas of the site does it impact?

This has no direct impact on the user experience as it is a background job. However, it does impact the Sign Up Service given that each retry represents an additional request to the Sign Up Service to update whether a veteran has accepted or rejected the Terms of Use.

## Acceptance criteria


## Requested Feedback

Refer to ToU epic for additional context on this effort.
